src/components/atoms/UserAvatar/UserAvatar.tsx

```diff
@@ -40,6 +40,13 @@ const AVATAR_SIZE_MAP: { [key in UserAvatarSize]: number | null } = {
   full: null,
 };
 
+const onProfileImageLoadError: React.ReactEventHandler<HTMLImageElement> = ({
+  currentTarget,
+}) => {
+  currentTarget.onerror = null; // prevents looping
+  currentTarget.src = DEFAULT_PROFILE_IMAGE;
```

Comment:
@sunny-viktoryia please check this PR https://github.com/sparkletown/sparkle/pull/2727 before using the constant.
I was hoping @mike-lvov would approve the PR since his ask can be done in a subsequent one

Reply:
I see that in your PR you return an image without checking if the image exists. 
![image](https://user-images.githubusercontent.com/63313732/149094770-147e19d4-868c-4396-8da4-8ab0849a86f0.png)

Mine works for the broken links. Let's say https://google.com/broken-image.png - if the image doesn't exist it'll be replaced by the default picture. 

Probably, we can combine both approaches. 🙃

Reply:
Please use `determineAvatar()` instead of the constant

Reply:
and upgrade it if necessary to solve your use case

Reply:
@sunny-viktoryia do you plan to use that `determineAvatar` or proceed without it?
It's used in many places in our app, so probably would be a good idea to extend it's logic

Reply:
I do. But haven't had enough time for that so far. 😬

Reply:
I have updated `determineAvatar` function and made it return an `onLoadError` function.
So now you can get
```ts
const [src, onError] = determineAvatar(options);
...
<img src={src} onError={onError}>
```
`onError` function chooses the fallback image in case the `src` does not exist.

The default pictures now are different - you can check the screenshot attached to the description. 

<hr>

My first approach was to check if the avatar exists in the `determineAvatar` directly, but those changes came with some inconveniences:
* the function would become `async`, which would complicate its usage
* we would get the avatar several times, which could affect our performance in the auditorium, for example, where we have to check many of avatars

@alexey-petrov @goran-peoski-work the PR is ready for re-review. 🙃 will be happy to hear your feedback once you have free time.

Reply:
This looks like a good idea, I might just request we switch from an array to an object, since that will give us more flexibility for backwards and forwards compat down the road. 

The idea is: a function that takes only one object as an argument and returns only one object as a result can just add more fields and/or ignore fields to those objects without requiring us to go over the project and update every single use.